### ISSUE TITLE
Is it possible to use a binary appkey.key file?

### ISSUE THREAD

**Issue by shaunmulligan** [bzf/rustify#8]
It seems spotify has stopped giving out keys, so I can't get a C-code key, but luckily I still have my binary appkey.key file from a previous project. I was wondering if there was an easy way I could use this with rustify?

**Comment by bzf** [comment]
Good question, I'm not sure!
Another approach would be trying to convert your binary file to a C array. A quick search and I found [this](http://unix.stackexchange.com/questions/176111/how-to-dump-a-binary-file-as-a-c-c-string-literal#176112).

Let me know if it works or I'll look more into it! 😄

**Comment by shaunmulligan** [comment]
Hey @bzf thanks for the idea. I gave it a try, but it seems to throw a `Segmentation fault: 11` at runtime, on OSX 10.11.5. After a quick google, apparently this is usually due to an invalid appkey (https://github.com/FrontierPsychiatrist/node-spotify/issues/70). However, I tested my appkey.key against pyspotify to check that it's still valid and that works.

Perhaps I am doing something stupid, or maybe its not as simple as reading in the binary file. This is what my application_key.rs looks like:

```
// examples/application_key.rs
use std::fs::File;
use std::io::Read;

pub fn get() -> Vec<u8> {
    let mut file=File::open("/Users/hobochild/Desktop/rustify/examples/spotify_appkey.key").unwrap();
    let mut buffer = Vec::new();
    file.read_to_end(&mut buffer).unwrap();
    return buffer;
}
```

**Comment by bzf** [comment]
Hmm, all of my previous projects that also uses this library seem to segfault as well.

Using a debugger it seems to crash on the `sp_session_create` call. Does it happen to you at the same place?

**Comment by shaunmulligan** [comment]
Yup, exactly the same place. Perhaps Spotify just switched off that part of the API, so no one can authenticate and create sessions anymore with libspotify? 

Which version of libspotify are you using in your tests?

**Comment by bzf** [comment]
I'm using the `API_VERSION 12` and version`12.1.51` of `libspotify`.

Compiling and running a small C program which calls the `sp_session_create` function doesn't seem to segfault, maybe it's something that's changed in Rust and raw pointers since I last worked with this?

I'm not going to have any time the coming weeks to look into this, but if you figure something out  and can get this working again in the meantime feel free to submit a PR. 😄

**Comment by shaunmulligan** [comment]
@bzf it seems you are correct. I tested against rust 1.12.0 and it works without issue, so something in 1.13.0 is causing the problem. I will try have a look into it, but I must admit, I am very new to the world of rust, so chances are it won't be very fruitful in the short term. Thanks for the help, I guess we can consider this issue closed, as the work around for the appkey.key seems to work. I have created a new issue for the incompatibility with 1.13.0.
